opengovsg/checkfirst
# feat: login page and frontend authentication

## Solution

Add login page and frontend authentication. 

## Before & After Screenshots
<img src='https://user-images.githubusercontent.com/3666479/104885293-c8995c80-59a2-11eb-85e4-e02c90a1ca94.png' width=650 />
<img src='https://user-images.githubusercontent.com/3666479/104885316-d7800f00-59a2-11eb-8659-e6f72b793215.png' width=650 />

## Tests
- Login with a valid email
- Login without a valid email
- Resend OTP
- `/projects` should only be available after login

**New dependencies**:
- `react-query` : Hooks for fetching, caching and updating asynchronous data in React
- `axios` : Library for making HTTP calls

## TODOs
- Need `/logout` route
- Need `/user/:id` route to check if user is already logged in
- Use design from Natalie
